src/pylibsshext/scp.pyx

```diff
@@ -0,0 +1,132 @@
+#
+# This file is part of the pylibssh library
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see file LICENSE.rst in this
+# repository.
+
+import os
+
+from cpython.bytes cimport PyBytes_AS_STRING
+from cpython.mem cimport PyMem_Malloc, PyMem_Free
+
+from pylibsshext.errors cimport LibsshSCPException
+from pylibsshext.session cimport get_libssh_session
+
+
+cdef class SCP:
+    def __cinit__(self, session):
+        self.session = session
+        self._libssh_session = get_libssh_session(session)
+
+    def __dealloc__(self):
+        if self._libssh_session is not NULL:
+            self._libssh_session = NULL
```

Comment:
Assigning NULL if memory is allocated doesn't seem right, might result in memory leaks. Is the `__dealloc__` required here? The deallocation of `self._libssh_session` should be taken care of by the session object.